cmp/grid/impl/ColumnWidthCalculator.js

```diff
@@ -88,19 +92,35 @@ export class ColumnWidthCalculator {
     }
 
     calcLevelWidth(gridModel, records, column, options, indentationPx = 0) {
-        const {field, getValueFn, renderer, cellClassFn, cellClassRules} = column,
+        const {field, getValueFn, renderer, rendererIsComplex, cellClassFn, cellClassRules} = column,
             {store, sizingMode, rowClassFn, rowClassRules} = gridModel,
             bufferPx = column.autosizeBufferPx ?? options.bufferPx;
 
         // 1) Get Map of rendered values to List of records that contain them
-        const recsByValue = new Map();
+        const recsByValue = new Map(),
+            renderedValues = new Map();
+
         records.forEach(record => {
             if (!record) return;
+            if (!renderedValues.size >= this.VALUE_SAMPLES && !record.isSummary) return;
```

Comment:
this logic seems wrong to me.  Why the initial '!' on line 105

Reply:
Typo!